charts/gardener-extension-admission-gcp/charts/runtime/templates/deployment.yaml

```diff
@@ -21,11 +21,8 @@ spec:
       labels:
 {{ include "labels" . | indent 8 }}
     spec:
-      {{- if .Values.global.kubeconfig }}
-      automountServiceAccountToken: false
-      {{- else }}
       serviceAccountName: {{ include "name" . }}
```

Comment:
I am not sure what do you mean by "optional service account". To my understanding if service account is not provided then the default for the namespace will be used ([ref](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server)). Knowing that wouldn't it be best for a workload to have its own dedicated serviceaccount even if it is not going to use its credentials to authenticate against the API server?

Reply:
There are three cases now: Component uses

1. a dedicated `ServiceAccount` from the runtime cluster to talk to the API server in the runtime cluster (no kubeconfig needed, no token projection needed)
2. a kubeconfig for the target cluster (no `ServiceAccount` needed, no token projection needed)
3. a generic kubeconfig and a projected token to talk to the target cluster

With your current PR/proposal, if I run my system in case 2. then I would still always get a `ServiceAccount` although I don't need it.

Reply:
If I understand you correctly in case 2. you prefer having assigned the `default` service account to the workload instead of having a dedicated serviceaccount object deployed in the cluster. Is this correct?

Reply:
No, in case 2 I would not want to have any service account at all (like it was before) since it is not needed.